src/couchdb/migration.controller.ts

```diff
@@ -1,81 +1,69 @@
+    private couchdbService: CouchdbService,
     private configService: ConfigService,
   ) {}
 
   @ApiOperation({
-    description: 'Transform site-settings config from the central config doc into its own SiteSettings entity.',
+    description:
+      'Transform site-settings config from the central config doc into its own SiteSettings entity.',
   })
   @Post('site-settings')
   async createSiteSettings() {
-    const results = {};
-    for (const { name, password } of credentials) {
-      await this.migrateUiConfigToSettings(name, password)
-        .then((res) => (results[name] = res))
-        .catch((err) =>
-          console.error('ERROR migrating UiConfig for: ' + name, err),
-        );
-    }
-    return results;
+    return await this.couchdbService.runForAllOrgs(
+      credentials,
+      async (couchdb: Couchdb) => {
+        return this.migrateUiConfigToSettings(couchdb);
+      },
+    );
   }
 
-  private async migrateUiConfigToSettings(name, password) {
+  private async migrateUiConfigToSettings(couchdb: Couchdb) {
     const configPath = '/app/Config:CONFIG_ENTITY';
-    const config = await this.couchDB.get(name, configPath, password);
+    const config = await couchdb.get(configPath);
     const uiConfig = config.data.appConfig as UiConfig;
     const siteSettings = {
       siteName: uiConfig.site_name,
       displayLanguageSelect: uiConfig.displayLanguageSelect,
       language: uiConfig.default_language,
     };
     if (uiConfig.logo_path) {
-      await this.uploadLogo(uiConfig.logo_path, name, password)
+      await this.uploadLogo(uiConfig.logo_path, couchdb)
         .then((fileName) => (siteSettings['logo'] = fileName))
         .catch((err) => console.error('ERROR uploading logo', err));
     }
-    await this.couchDB.put(
-      name,
-      '/app/SiteSettings:global',
-      siteSettings,
-      password,
-    );
+    await couchdb.put('/app/SiteSettings:global', siteSettings);
     // delete deprecated attribute
     delete config.data.appConfig;
-    await this.couchDB.put(name, configPath, config, password);
+    await couchdb.put(configPath, config);
     console.log('migrated', name);
     return siteSettings;
   }
 
-  private async uploadLogo(logoPath: string, name, password) {
+  private async uploadLogo(logoPath: string, couchdb: Couchdb) {
     logoPath = logoPath.startsWith('/') ? logoPath : '/' + logoPath;
-    const url = `https://${name}.${this.domain}${logoPath}`;
+    const url = `https://${couchdb.org}.${this.domain}${logoPath}`;
     const logo = await fetch(url).then((res) => res.blob());
     const logoBuffer = await logo.arrayBuffer();
     const resizedLogo = await sharp(logoBuffer)
       .resize({ width: 300 })
       .toBuffer();
-    const resizedBlob = new Blob([new Uint8Array(resizedLogo)], {
+    const resizedBlob = new Blob([new Uint8Array(resizedLogo as any)], {
       type: logo.type,
     });
     const attachmentsPath = '/app-attachments/SiteSettings:general';
```

Comment:
```suggestion
    const attachmentsPath = '/app-attachments/SiteSettings:global';
```
Good find that this was incorrect.